lib/nerves_network/dhcp_manager.ex

```diff
@@ -277,19 +266,47 @@ defmodule Nerves.Network.DHCPManager do
     state
   end
 
-  @spec configure(t, Types.udhcp_info) :: t
+  defp setup_iface(state, info) do
+    case Nerves.NetworkInterface.setup(state.ifname, info) do
+      :ok -> :ok
+      {:error, :eexist} -> :ok
+        #It may very often happen that at the renew time we would receive the lease of the very same IP address...
+        #In such a case whilst adding already existent IP address to the network interface we shall receive 'error exists'.
+        #It definitely is non-critical situation and actually confirms that we do not have to take any action.
+    end
+  end
+
+  defp remove_old_ip(state, info) do
+    old_ip = info[:"old_ipv4_address"] || ""
+    new_ip = info[:"ipv4_address"] || ""
+
+    if old_ip == "" or new_ip == old_ip do
+      :ok
+    else
+      Logger.debug fn -> "Removing ipv4 address = #{inspect old_ip} from #{inspect state.ifname}" end
+      Nerves.NetworkInterface.setup(state.ifname, %{:"-ipv4_address" => old_ip})
+    end
+  end
+
+  defp configure(:no_resolv_conf, state, info) do
+    remove_old_ip(state, info)
+    :ok = setup_iface(state, info)
+    state
+  end
+
   defp configure(state, info) do
-    Logger.debug("DHCP state #{inspect state} #{inspect info}")
+    Logger.warn("DHCP state #{inspect state} #{inspect info}")
 
-    :ok = Nerves.NetworkInterface.setup(state.ifname, info)
+    :ok = setup_iface(state, info)
     :ok = Nerves.Network.Resolvconf.setup(Nerves.Network.Resolvconf, state.ifname, info)
+
+    # Show that the route has been updated
+    System.cmd("route", []) |> elem(0) |> Logger.error
```

Comment:
Should this be an error log? That feels like not an error to me.

Do we care to wait on the command before stopping the configure? If we don't we should put the call into an anonymous function. `Logger.error fn-> System.cmd("route", []) |> elem(0)`

Reply:
It should not but once one tried to enable a debug print using the Ring Logger one would UNDERSTAND!! This is my fault as I must have used Logger.error in the original dhclient Alan re-used... I still have it on my ToDo list the RingLogger improvements / jira items' creation...